### ISSUE TITLE
[Bug]: webDarkTheme <Select appearance="underline"> dropdown is low contrast on Windows

### ISSUE THREAD

**Issue by winmill** [microsoft/fluentui#29912]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 5.22 GB / 15.95 GB
  Browsers:
    Edge: Spartan (44.19041.3636.0), Chromium (119.0.2151.72)
    Internet Explorer: 11.0.19041.3636
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/p/sandbox/determined-shirley-tp42h7

## Actual Behavior

In dark mode when `<Select>` is used with `appearance="underline"`, the dropdown options are white on gray background on Windows.  Mac and iPhone not impacted.

This can be seen on the FluentUI React site at:
https://react.fluentui.dev/iframe.html?viewMode=docs&id=components-select--default&globals=storybook_fluentui-react-addon_theme:web-dark#appearance

![Select](https://github.com/microsoft/fluentui/assets/32454612/1ef1dac3-cffe-439d-82a8-33d614af4ac0)

## Expected Behavior
Dropdown options should be on dark background, like in `<Select appearance="outline">`, `<Combobox>` or `<Dropdown>`.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by khmakoto** [comment]
I can confirm this is still the case.

**Comment by dmytrokirpa** [comment]
Fixed by #31736
